pkg/asset/agent/image/baseiso.go

```diff
@@ -88,26 +97,58 @@ func getIsoFromReleasePayload() (string, error) {
 // Dependencies returns dependencies used by the asset.
 func (i *BaseIso) Dependencies() []asset.Asset {
 	return []asset.Asset{
+		&manifests.AgentManifests{},
 		&agent.OptionalInstallConfig{},
+		&mirror.RegistriesConf{},
 	}
 }
 
 // Generate the baseIso
-func (i *BaseIso) Generate(p asset.Parents) error {
+func (i *BaseIso) Generate(dependencies asset.Parents) error {
 
+	log := logrus.New()
 	// TODO - if image registry location is defined in InstallConfig,
 	// ic := &agent.OptionalInstallConfig{}
 	// p.Get(ic)
+
 	// use the GetIso function to get the BaseIso from the release payload
-	isoGetter := newGetIso(GetIsoPluggable)
-	baseIsoFileName, err := isoGetter.getter()
-	if err != nil {
-		return errors.Wrap(err, "failed to get base iso image")
+	agentManifests := &manifests.AgentManifests{}
+	dependencies.Get(agentManifests)
+
+	var baseIsoFileName string
+	if agentManifests.ClusterImageSet != nil {
+		releaseImage := agentManifests.ClusterImageSet.Spec.ReleaseImage
+		pullSecret := agentManifests.GetPullSecretData()
+		registriesConf := &mirror.RegistriesConf{}
+		dependencies.Get(agentManifests, registriesConf)
+
+		// If we have the image registry location and 'oc' command is available then get from release payload
+		ocRelease := NewRelease(&executer.CommonExecuter{},
+			Config{MaxTries: OcDefaultTries, RetryDelay: OcDefaultRetryDelay})
+
+		log.Info("Extracting base ISO from release payload")
+		baseIsoFileName, err := ocRelease.GetBaseIso(log, releaseImage, pullSecret, registriesConf.MirrorConfig, archName)
+		if err == nil {
+			log.Debugf("Extracted base ISO image %s from release payload", baseIsoFileName)
+			i.File = &asset.File{Filename: baseIsoFileName}
+			return nil
+		}
+		if !errors.Is(err, &exec.Error{}) { // Already warned about missing oc binary
+			log.Warning("Failed to extract base ISO from release payload - check registry configuration")
+		}
 	}
 
-	i.File = &asset.File{Filename: baseIsoFileName}
+	log.Info("Downloading base ISO")
+	isoGetter := newGetIso(GetIsoPluggable)
+	baseIsoFileName, err2 := isoGetter.getter()
+	if err2 == nil {
+		log.Debugf("Using base ISO image %s", baseIsoFileName)
+		i.File = &asset.File{Filename: baseIsoFileName}
+		return nil
+	}
+	log.Debugf("Failed to download base ISO: %s", err2)
 
-	return nil
+	return errors.Wrap(err2, "failed to get base ISO image")
```

Comment:
This was intended to be `err`, not `err2` (see first paragraph of https://github.com/openshift/installer/pull/6154/files#r927768755)